components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaAuthMgtDAOImpl.java

```diff
@@ -0,0 +1,534 @@
+                prepStmt.setString(1, cibaAuthCodeDOKey);
+
+                ResultSet resultSet = prepStmt.executeQuery();
+                if (resultSet.next()) {
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully obtained authenticationStatus of TokenRequest  with " +
+                                "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+                    }
+                    return resultSet.getString(1);
+
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("No field found for authenticationStatus of TokenRequest  with " +
+                                "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+                    }
+                    return null;
+                }
+            }
+        } catch (SQLException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred in obtaining authenticationStatus of TokenRequest  with " +
+                        "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+            }
+            throw new CibaCoreException(ErrorCodes.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Returns the authenticated user of authenticationRequest for cibaAuthCodeDOKey.
+     *
+     * @param cibaAuthCodeDOKey identifier of CibaAuthCode
+     * @return Returns AuthenticatedUser.
+     * @throws CibaCoreException Exception thrown from CibaCore Component.
+     */
+    public String getAuthenticatedUser(String cibaAuthCodeDOKey) throws CibaCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.CibaSQLQueries.RETRIEVE_AUTHENTICATED_USER)) {
+                prepStmt.setString(1, cibaAuthCodeDOKey);
+
+                ResultSet resultSet = prepStmt.executeQuery();
+                if (resultSet.next()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully obtained authenticatedUser of TokenRequest  with " +
+                                "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+                    }
+                    return resultSet.getString(1);
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("No field found for authenticatedUser of TokenRequest  with " +
+                                "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+                    }
+                    return null;
+                }
+            }
+        } catch (SQLException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred in obtaining authenticatedUser of TokenRequest  with " +
+                        "cibaAuthCodeDOKey : " + cibaAuthCodeDOKey);
+            }
+            throw new CibaCoreException(ErrorCodes.INTERNAL_SERVER_ERROR, e.getMessage());
```

Comment:
All the CibaCoreException exceptions have the same error code 'ErrorCodes.INTERNAL_SERVER_ERROR'. So is it required to set the error code?